packages/audio-block/src/AudioBlock.tsx

```diff
@@ -0,0 +1,201 @@
+import 'tailwindcss/tailwind.css';
+import { BlockSettings, TextPosition } from './types';
+import { AUDIO_EXTENSIONS, AUDIO_ID } from './settings';
+import { UploadPlaceholder } from './components/UploadPlaceholder';
+import { ItemToolbar } from './components/ItemToolbar';
+import { BlockAttachments } from './components/BlockAttachments';
+import { useEffect, useState } from 'react';
+
+const DEFAULT_CONTENT_TITLE = '[{"type":"heading3","children":[{"text":""}]}]';
+const DEFAULT_CONTENT_DESCRIPTION = '[{"type":"paragraph","children":[{"text":""}]}]';
+
+const customTitlePlugins = new PluginComposer();
+customTitlePlugins
+    .setPlugin([new InitPlugin(), new TextStylePlugin() as Plugin<PluginProps>])
+    .setPlugin([new BoldPlugin(), new ItalicPlugin(), new UnderlinePlugin(), new StrikethroughPlugin()])
+    .setPlugin([new AlignLeftPlugin(), new AlignRightPlugin(), new AlignCenterPlugin(), new AlignJustifyPlugin()])
+    .setPlugin([new ResetFormattingPlugin()]);
+
+export const AudioBlock = ({ appBridge }: BlockProps) => {
+    const [isLoading, setIsLoading] = useState(false);
+    const [droppedFiles, setDroppedFiles] = useState<FileList | null>(null);
+    const isEditing = useEditorState(appBridge);
+    const [blockSettings, setBlockSettings] = useBlockSettings<BlockSettings>(appBridge);
+    const { blockAssets, deleteAssetIdsFromKey, updateAssetIdsFromKey } = useBlockAssets(appBridge);
+    const [openFileDialog, { selectedFiles }] = useFileInput({ multiple: false, accept: 'audio/*' });
+    const { designTokens } = useGuidelineDesignTokens();
+    const { title, description } = blockSettings;
+    const audio = blockAssets?.[AUDIO_ID]?.[0];
+
+    const [uploadFile, { results: uploadResults, doneAll }] = useAssetUpload({
+        onUploadProgress: () => !isLoading && setIsLoading(true),
+    });
+
+    const audioBlockClassNames = joinClassNames([
+        'tw-flex tw-flex-col tw-gap-3',
+        blockSettings.positioning === TextPosition.Above && 'tw-flex-col-reverse',
+    ]);
+
+    const audiotTagClassNames = joinClassNames([
+        'tw-w-full tw-mt-5',
+        isEditing && 'group-hover:tw-border group-hover:tw-border-box-selected-inverse group-hover:tw-rounded-[4px]',
+    ]);
+
+    const saveTitle = (value: string) =>
+        value !== blockSettings.title &&
+        setBlockSettings({
+            title: value,
+        });
+
+    const saveDescription = (value: string) =>
+        value !== blockSettings.description &&
+        setBlockSettings({
+            description: value,
+        });
+
+    const onRemoveAsset = () => {
+        deleteAssetIdsFromKey(AUDIO_ID, [audio?.id]);
+    };
+
+    const updateAudioAsset = async (audio: Asset) => {
+        updateAssetIdsFromKey(AUDIO_ID, [audio.id]);
+        setIsLoading(false);
+    };
+
+    const openAssetChooser = () => {
+        appBridge.openAssetChooser(
+            async (result) => {
+                updateAudioAsset(result[0]);
```

Comment:
maybe set IsLoading to true here as well. I know it's snappier than upload but still takes some time

Reply:
you are right... 
Fixed within [Commit](https://github.com/Frontify/guideline-blocks/commit/3c5b13c9099812d0a1d2e4f409e60e159d767cb0)